### ISSUE TITLE
ctt receiver registration api

### ISSUE THREAD

**Issue by denislepage** [MotusDev/Motus-TO-DO#509]
With CTT, we discussed the notion of being able to register a receiver through the Motus API. Ideally, CTT would push those when they are ordered by users.

This is mainly to avoid having users trying to read the serial numbers manually.

Can we set up an API entry to assign a new receiver to a project? This is almost the same we already have for tag finder, I assume. It'll likely require a serial number, a project number, and user authentication like we proposed for tags. The permissions would here be checked against whether users are permitted to manage receivers instead of tags.

We are not planning to allow passing other metadata details through the API at this point. Users would sign into motus and see the list of receivers for which they could create new deployments.

We will need to support new regex for CTT base station serial numbers. Bob should be finalizing this shortly. We're probably looking at a short prefix + 15 characters (without any specific pattern, from what I could tell).

**Comment by pmorrill** [comment]
If Catherine can do her magic (?) to create a recv. registration procedure, I can set this up. Of course, we will need the regex for the final version, but we could build the framework ahead of time.

**Comment by cjardine-bsc** [comment]
Paul, will you be passing the proc an email or a resolved user_id?

**Comment by pmorrill** [comment]
I was hoping for the same input as you used for prAPIImportTagsCTT. That requires an email only (plus project id), plus all the specific stuff for the tag (recv in this case)

**Comment by cjardine-bsc** [comment]
Bipity Bopity BOO!

 prAPIImportSensorsCTT, prAPIImportSensorsTransfer

**Comment by cjardine-bsc** [comment]
Okay, I simplified. You should be able to just use prAPIImportSensorsCTT 

    @import_id int OUTPUT,
    @projectID int,
    @receiverID varchar(20),
    @ts float,
    @userID int,  
    @receiverType varchar(15), 
    @code varchar(20) OUTPUT,
    @deviceID int OUTPUT,
    @sensorID int OUTPUT

For tags, you can do the same. Get rid of the prAPIImportTagsTransfer and just use prAPIImportTagsCTT. It will now expect a userID instead of an email. 

	@import_id int OUTPUT,
    @projectID int,
    @tag_code char(8),
    @type varchar(12) = '32-BIT',
    @codeSet varchar(32) = 'Digital',
    @manufacturer varchar(32) = 'CTT',
    @model varchar(50) = 'LIFETAG1',
    @nomFreq float,
    @period float,
    @userID varchar(255), 
    @ts float,
    @tag_id int OUTPUT,
    @code varchar(20) OUTPUT
   

Let me know if you'd like me to tweak the order of the output parameters.

**Comment by pmorrill** [comment]
I have added the support for registering recv's. I also adjusted the tag registration handler so it uses the userId as input, rather than email address.

However, when I try to **re-register** a recv, the proc does not seem to be setting the '@code' parameter, and an exception is thrown when I try to retrieve that parameter.

Otherwise, seems to be working. I will update the documentation for Bob / CTT
